### PR TITLE
MODLOGSAML-71

### PR DESCRIPTION
**Pull request by MikhailFokanov**
- Allow to upload the IdP metadata to mod-login-saml.
- Check that IdP responses are signed by the IdP public key.

### CODE REVIEW

File: src/test/java/org/folio/rest/impl/SamlAPITest.java [hunk 1]
```diff
@@ -498,6 +505,33 @@ public void putConfigurationEndpoint(TestContext context) {
...
+    given()
+      .header(TENANT_HEADER)
+      .header(TOKEN_HEADER)
+      .header(OKAPI_URL_HEADER)
+      .header(JSON_CONTENT_TYPE_HEADER)
+      .body(jsonString)
+      .put("/saml/configuration")
+      .then()
+      .statusCode(200)
+      .body(matchesJsonSchemaInClasspath("ramls/schemas/SamlConfig.json"));
```
Comment (julianladisch) [inline src/test/java/org/folio/rest/impl/SamlAPITest.java:532]: Can you extend the tests by sending a response to mod-login-saml and check that a response signed with correct certificate is accepted and a response signed with wrong certificate is rejected?

**Review by julianladisch — CHANGES_REQUESTED** [review]

File: src/test/java/org/folio/rest/impl/SamlAPITest.java (hunk 1 above)
Comment (MikhailFokanov) [reply src/test/java/org/folio/rest/impl/SamlAPITest.java:532]: Is there any example?

File: ramls/schemas/SamlConfigRequest.json [hunk 2]
```diff
@@ -23,6 +23,11 @@
       "type": "string",
       "required": true
     },
+    "idpMetadata": {
+      "description": "the SAML attribute to use for matching against a FOLIO user",
```
Comment (julianladisch) [inline ramls/schemas/SamlConfigRequest.json:27]: Can you fix the description?

File: src/test/java/org/folio/rest/impl/SamlAPITest.java (hunk 1 above)
Comment (julianladisch) [reply src/test/java/org/folio/rest/impl/SamlAPITest.java:533]: You might try some reference environment with SSO enabled and copy&paste the response from the browser developer tools.

File: ramls/schemas/SamlConfigRequest.json [hunk 3]
```diff
@@ -23,6 +23,11 @@
       "type": "string",
       "required": true
     },
+    "idmXml": {
```
[the remark this answers is not in the document]
Comment (MikhailFokanov-ebsco) [reply ramls/schemas/SamlConfigRequest.json]: done

File: src/test/java/org/folio/rest/impl/SamlAPITest.java [hunk 4]
```diff
@@ -308,6 +312,78 @@ public void loginCorsTests() throws IOException {
...
+
+    given()
+      .header(new Header(HttpHeaders.ORIGIN.toString(), origin))
+      .header(TENANT_HEADER)
+      .header(TOKEN_HEADER)
+      .header(OKAPI_URL_HEADER)
+      .header(ACCESS_CONTROL_REQ_HEADERS_HEADER)
+      .header(ACCESS_CONTROL_REQUEST_METHOD_HEADER)
+      .body(readResourceToString("saml_response.txt"))
+      .options("/saml/callback")
```
Comment (julianladisch) [inline src/test/java/org/folio/rest/impl/SamlAPITest.java:332]: This should be a POST request, not a OPTIONS pre-flight request.

**Review by adamdickmeiss — APPROVED** [review]

**Review by julianladisch — APPROVED** [review]
Thanks, looks good to me.
More tests need to be added in a separate pull request.

**Pull request merged**